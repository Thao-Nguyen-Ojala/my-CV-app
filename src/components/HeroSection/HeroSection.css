/*color: #276afb;*/
.hero-section {
  display:flex; 
  justify-content:center;
}

.home__hero-section {
  background-color: #fff;
}

.home__hero-row {
  align-items: center;
}

.row {
  display: flex;
  margin: 0 -1.5rem -1.5rem -1.5rem;
  flex-wrap: wrap;
  align-content: stretch;
  max-width: 120rem;
  margin: auto;
}
.col {
  margin: 3rem 0 2rem 0;
  padding: 0 1.5rem 0 1.5rem;
  flex: 1;
  max-width: 50%;
  flex-basis: 50%;
}

.home__hero-text-wrapper {
  max-width: 100%;
  margin-top: 5rem;
  padding: 0 0 12rem 0;
}

.top-line {
  color: #f00946;
  font-size: 1.5rem;
  line-height: 1.5rem;
  font-weight: 700;
  letter-spacing: 1.4px;
  text-transform: uppercase;
  margin-bottom: 1.5rem;
}

.heading {
  color: #276AFB;
  margin-bottom: 1.5rem;
  font-size: 3rem;
  line-height: 1.1;
  font-weight: 600;
  text-align: left;
}

.darkBg {
  background-color: #242634;
}

.home__hero-subtitle {
  max-width: 100%;
  font-size: 1.6rem;
  line-height: 2rem;
  text-align: justify;
  margin-left: 0;
  color: #f7f8fa;
  white-space: pre-line;
}

.dark {
  color: #242634;
}

.home__hero-button {
  padding: 24px 0 0 0;
}

.home__hero-img-wrapper {
  max-width: 100%;
}

.home__hero-img {
  max-width: 80%;
  margin: auto;
  margin-top: 5rem;
  display: block;
}

img {
  border: 0;
  display: inline-block;
  position: relative;
}

@media screen and (max-width: 991px) {
  .container {
    padding-right: 3rem;
    padding-left: 3rem;
  }
}

@media screen and (max-width: 768px) {
  .row {
    max-width: 100%;
  }

  .col {
    max-width: 100%;
    flex-basis: 100%;
    padding: 0;
    margin-top: 1rem ;
  }

  .home__hero-text-wrapper {
    max-width: 100%; 
    margin-bottom: 0;
    padding: 0 0 2rem 0 ;
  }

  .top-line {
    margin: 2rem 0 2rem 0;
    text-align: center;
  }

  .heading {
    margin: 1rem 0 1rem 0;
    text-align: center;
  }

  .home__hero-subtitle {
    max-width: 100%;
    margin: 1rem 0 1rem 0;
    text-align: justify;
  }

  .home__hero-button {
    padding: 24px 0 0 0;
    margin: 10px 0 0 0;
  }

  .home__hero-img {
    max-width: 80%;
    margin-bottom: 1rem;
    display: block;
  }

}

@media screen and (max-width: 376px) {
  .container {
    padding: 0;
  }

  .row {
    max-width: 100%;
    margin: 0;
  }

  .col {
    margin: 0;
    max-width: 100%;
    flex-basis: 100%;

  }

  .home__hero-text-wrapper {
    max-width: 100%;
    padding: 0
  }

  .top-line {
    margin: 2rem 0 2rem 0;
    text-align: center;
    font-size: 1.5rem;
  }

  .heading {
    margin: 1rem 0 1rem 0;
    text-align: center;
    font-size: 2.2rem;
  }

  .home__hero-subtitle {
    max-width: 100%;
    margin: 0;
    font-size: 1.5rem;
    text-align: justify;
  }

  .home__hero-button {
    padding: 24px 0 0 0;
    margin: 10px 0 0 0;
  }

  .home__hero-img {
    margin: auto;
    padding: 1rem 0 1rem 0;
    display: block;
  }

}