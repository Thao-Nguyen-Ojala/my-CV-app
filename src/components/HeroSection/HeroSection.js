import React from 'react'
import { Button } from '..'
import './HeroSection.css'

function HeroSection({
  lightBg, 
  topLine, 
  lightTextDesc, 
  headline, 
  description, 
  buttonLabel,
  buttonLabel2, 
  img, 
  alt, 
  imgStart,
  url,
  url2,
  name
}) {
  return (
    <>
      <div className='hero-section' id={name}>
        <div 
          className={lightBg ? 'home__hero-section' : 'home__hero-section darkBg'}
        >
          <div className='container'>
            <div className='row home__hero-row'
              style={{display: 'flex', flexDirection: imgStart === 'start' ? 'row-reverse' : 'row'}}
            >
              <div className='col'>
                <div className='home__hero-text-wrapper'>
                  {topLine && <div className='top-line'>{topLine}</div>}
                  <h1 className='heading'>{headline}</h1>
                  <p className={lightTextDesc ? 'home__hero-subtitle' : 'home__hero-subtitle dark'}>{description}</p>
                  {url && <div className='home__hero-button'>
                    <Button buttonSize='btn--wide' buttonColor='blue' onClick={() => {window.open(url, '_blank')}}>{buttonLabel} </Button>
                  </div>}
                  {url2 && <div className='home__hero-button'>
                    <Button buttonSize='btn--wide' buttonColor='blue' onClick={() => {window.open(url2, '_blank')}}>{buttonLabel2} </Button>
                  </div>}
                </div>
              </div>

              <div className='col'>
                <div className='home_hero-img-wrapper'>
                  <img src={img} alt={alt} className='home__hero-img' />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  )
}

export default HeroSection

