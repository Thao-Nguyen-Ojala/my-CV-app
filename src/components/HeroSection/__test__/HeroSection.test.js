import React from 'react';
import ReactDOM from 'react-dom';
import HeroSection from '../HeroSection';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

afterEach(cleanup)

it('render HeroSection without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Router>
    <HeroSection>HeroSection testing</HeroSection>
    </Router>, div)
})

it('render HeroSection topline correctly', () => {
  const {getByTestId} = render(<HeroSection topLine={'Testing topLine'}></HeroSection>)
  expect(getByTestId('heroSection-test')).toHaveTextContent('Testing topLine')
})

it('render HeroSection headline correctly', () => {
  const {getByTestId} = render(<HeroSection headline={'Testing headline'}></HeroSection>)
  expect(getByTestId('heroSection-test')).toHaveTextContent('Testing headline')
})

it('render HeroSection description correctly', () => {
  const {getByTestId} = render(<HeroSection description={'Testing description'}></HeroSection>)
  expect(getByTestId('heroSection-test')).toHaveTextContent('Testing description')
})


it('matches HeroSection snapshot', () => {
  const testingHeroSection= renderer.create(<Router><HeroSection></HeroSection></Router>).toJSON();
  expect(testingHeroSection).toMatchSnapshot()
})