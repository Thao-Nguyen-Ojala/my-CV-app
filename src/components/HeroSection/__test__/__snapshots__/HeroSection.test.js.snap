// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches HeroSection snapshot 1`] = `
<div
  className="hero-section"
  data-testid="heroSection-test"
>
  <div
    className="home__hero-section darkBg"
  >
    <div
      className="container"
    >
      <div
        className="row home__hero-row"
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <div
          className="col"
        >
          <div
            className="home__hero-text-wrapper"
          >
            <h1
              className="heading"
            />
            <p
              className="home__hero-subtitle dark"
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="home_hero-img-wrapper"
          >
            <img
              className="home__hero-img"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
